src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebApplicationTests.cs

```diff
@@ -232,6 +233,66 @@ public void WebApplicationBuilderWebHostSettingsThatAffectTheHostCannotBeModifie
             }));
         }
 
+        [Fact]
+        public void WebApplicationBuilderHostEnvironmentAndWebHostEnvironmentAreTheSame()
+        {
+            var projectName = typeof(WebApplicationTests).Assembly.GetName().Name;
+            var assemblyName = Assembly.GetEntryAssembly().GetName().Name;
+
+            var builder = WebApplication.CreateBuilder(new[] { $"--applicationName={assemblyName}" });
+
+            Assert.Equal(assemblyName, builder.Environment.ApplicationName);
+            builder.Host.ConfigureAppConfiguration((context, config) =>
+            {
+                Assert.Equal(assemblyName, context.HostingEnvironment.ApplicationName);
+            });
+
+            builder.WebHost.ConfigureAppConfiguration((context, config) =>
+            {
+                Assert.Equal(assemblyName, context.HostingEnvironment.ApplicationName);
+            });
+
+            var app = builder.Build();
+            var hostEnv = app.Services.GetRequiredService<IHostEnvironment>();
+            var webHostEnv = app.Services.GetRequiredService<IWebHostEnvironment>();
+
+            Assert.Equal(assemblyName, hostEnv.ApplicationName);
+            Assert.Equal(assemblyName, webHostEnv.ApplicationName);
+        }
+
+        [Fact]
+        public void WebApplicationBuilderCanFlowCommandLineConfigurationToApplication()
+        {
+            var builder = WebApplication.CreateBuilder(new[] { "--x=1", "--name=Larry", "--age=20", "--environment=Testing" });
+
+            Assert.Equal("1", builder.Configuration["x"]);
+            Assert.Equal("Larry", builder.Configuration["name"]);
+            Assert.Equal("20", builder.Configuration["age"]);
+            Assert.Equal("Testing", builder.Configuration["environment"]);
+            Assert.Equal("Testing", builder.Environment.EnvironmentName);
+
+            builder.WebHost.ConfigureAppConfiguration((context, config) =>
+            {
+                Assert.Equal("Testing", context.HostingEnvironment.EnvironmentName);
+            });
+
+            builder.Host.ConfigureAppConfiguration((context, config) =>
+            {
+                Assert.Equal("Testing", context.HostingEnvironment.EnvironmentName);
+            });
+
+            var app = builder.Build();
+            var hostEnv = app.Services.GetRequiredService<IHostEnvironment>();
+            var webHostEnv = app.Services.GetRequiredService<IWebHostEnvironment>();
+
+            Assert.Equal("Testing", hostEnv.EnvironmentName);
+            Assert.Equal("Testing", webHostEnv.EnvironmentName);
+            Assert.Equal("1", app.Configuration["x"]);
+            Assert.Equal("Larry", app.Configuration["name"]);
+            Assert.Equal("20", app.Configuration["age"]);
+            Assert.Equal("Testing", app.Configuration["environment"]);
+        }
```

Comment:
Both of these tests pass with the current `main` branch. Can we add regression tests?

Reply:
Not sure what you mean. I just wanted coverage that this still worked in case it gets broken

Reply:
I like the new tests. I just want to add at least one test that would have failed before this change that passes now so all the tests won't continue passing if we accidently revert the behavior by going back to a single ConfigurationManager or something.